lib/rspec/core/formatters/exception_presenter.rb

```diff
@@ -244,7 +244,7 @@ def formatted_message_and_backtrace(colorizer)
         end
 
         def exception_backtrace
-          exception.backtrace || []
+          RSpec.configuration.backtrace_formatter.format_backtrace(exception.backtrace) || []
```

Comment:
I added a cucumber scenario for it https://github.com/rspec/rspec-core/pull/2333/files#diff-494447daea98a33564df83824eacaad7R99

The problem is next
If I set `backtrace_exclusion_patterns` in config, I'm not expecting to see any details from this files: matching lines ignored in backtrace (working fine) and failure snippet displayed code on first not-ignored line (this is what I'm trying to fix)

With current functionality RSpec test defined in cucumber scenario having next output
 1) failure fails
     Failure/Error: raise unless arg =='123'
     RuntimeError:
     # ./spec/excluded_backtrace_paths_spec.rb:8:in `block (2 levels) in <top (required)>'

I think it's misleading because first backtrace line's file don't have displayed snippet and IMHO if someone wants to ignore file from backtrace, he/she most likely will be surprised to see code from ignored file